packages/replay/src/util/addEvent.ts

```diff
@@ -40,7 +40,7 @@ export async function addEvent(
   }
 
   try {
-    if (isCheckout) {
+    if (isCheckout && replay.recordingMode === 'buffer') {
```

Comment:
Is it possible we rely on this behavior in session mode as well?

e.g. when we call stopRecording, and then restart it?